layers/gre.go

```diff
@@ -154,11 +156,8 @@ func (g *GRE) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOpt
 		// zero out checksum bytes
 		buf[4] = 0
```

Comment:
Actually, I think the logic we want here should be the following, to allow users to add in their own checksums if they'd like:

```go
if g.ChecksumPresent {
  if opts.ComputeChecksums {
    g.Checksum = actuallyComptue()
  }
  // put in checksum regardless, allowing users to set their own checksum if they so desire.
  binary.BigEndian.PutUint16(buf[4:6], g.Checksum)
} else {
  // if not present, zero it out
  buf[4], buf[5] = 0, 0
}
```